components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -955,9 +967,33 @@ void updateConsumerApplication(OAuthConsumerAppDTO consumerAppDTO, boolean enabl
             oAuthAppDO.setRequestObjectEncryptionMethod(requestObjectEncryptionMethod);
             oAuthAppDO.setRequirePushedAuthorizationRequests(consumerAppDTO.getRequirePushedAuthorizationRequests());
             oAuthAppDO.setSubjectTokenEnabled(consumerAppDTO.isSubjectTokenEnabled());
-            oAuthAppDO.setSubjectTokenExpiryTime(consumerAppDTO.getSubjectTokenExpiryTime());;
+            oAuthAppDO.setSubjectTokenExpiryTime(consumerAppDTO.getSubjectTokenExpiryTime());
+            if (OAuth2Util.isJWTAccessTokenOIDCClaimsSeparationEnabled() &&
+                    oAuthAppDO.isJwtAccessTokenOIDCClaimsSeparationEnabled()) {
+                validateJwtAccessTokenClaims(consumerAppDTO, tenantDomain);
+                oAuthAppDO.setJwtAccessTokenClaims(consumerAppDTO.getJwtAccessTokenOIDCClaims());
+            }
         }
         dao.updateConsumerApplication(oAuthAppDO);
+        // We only trigger the jwt access token claims migration if the following conditions are met.
+        // 1. The new feature is enabled.
+        // 2. The app is not already in the new feature.
+        // 3. User update the app to enable the new feature.
+        if (OAuth2Util.isJWTAccessTokenOIDCClaimsSeparationEnabled() &&
+                !oAuthAppDO.isJwtAccessTokenOIDCClaimsSeparationEnabled() &&
+                consumerAppDTO.isJwtAccessTokenOIDCClaimsSeparationEnabled()) {
+            oAuthAppDO.setIsJwtAccessTokenOIDCClaimsSeparationEnabled(consumerAppDTO
+                    .isJwtAccessTokenOIDCClaimsSeparationEnabled());
+            // Add requested claims as jwt access token claims.
+            try {
+                addJwtAccessTokenClaims(oAuthAppDO, tenantDomain);
+                dao.updateConsumerApplication(oAuthAppDO);
```

Comment:
What is the point of updating the application inside the if block?
Can not we move https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2522/files#diff-6c1ece811bfc0f73bdb0d1718746f1f253119a01a268840ad862dcd3e9d077b5R977 to the end of the method?